plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/ConnectionEditor.java

```diff
@@ -223,8 +222,7 @@ private void disposeBindings() {
 	@Override
 	protected void onSelectedConnectionChanged(IObservableValue selectedConnection) {
 		IConnection conn = (IConnection) selectedConnection.getValue();
-		if(!(conn instanceof Connection)) return;
-		detailViewModel.setSelectedConnection((Connection)conn);
+		detailViewModel.setSelectedConnection(conn);
```

Comment:
yes, setSelectedConnection takes an IConnection argument, and the instance check prevents NewConnectionMarker instances to be set, which leads to the UI not being reset.